js/views/modals/purchase/Purchase.js

```diff
@@ -140,19 +166,51 @@ export default class extends BaseModal {
   }
 
   purchaseListing() {
+    // if the listing has shipping, set it
+    if (this.listing.get('shippingOptions') && this.listing.get('shippingOptions').length) {
+      if (this.shipping.selectedAddress) {
+        // set the address
+        this.order.addAddress(this.shipping.selectedAddress);
+      } else {
+        this.insertErrors(this.$shippingErrors, [app.polyglot.t('purchase.errors.missingAddress')]);
+      }
+    }
+
     // set the moderator
-    this.order.set('moderator', this.moderators.selectedIDs[0]);
-
-    $.post({
-      url: app.getServerUrl('ob/purchase'),
-      data: JSON.stringify(this.order.toJSON()),
-    })
-      .done((data) => {
-        console.log(data);
+    this.order.set({ moderator: this.moderators.selectedIDs[0] }, { validate: true });
+
+    if (!this.order.validationError) {
+      $.post({
+        url: app.getServerUrl('ob/purchase'),
+        data: JSON.stringify(this.order.toJSON()),
+        contentType: 'application/json',
       })
-      .fail((data) => {
-        console.log(data);
+        .done((data) => {
+          console.log(data);
+        })
+        .fail((data) => {
+          const response = data && data.responseText || '';
+          const errMsg = response ? JSON.parse(response).reason : '';
+          const errTitle = app.polyglot.t('purchase.errors.orderError');
+          openSimpleMessage(errTitle, errMsg);
+        });
+    } else {
```

Comment:
This code loops through any model errors, and either displays them in the appropriate container or displays them in a generic container.